gems/pending/disk/modules/VhdxDisk.rb

```diff
@@ -373,9 +374,13 @@ def region_table_entry(table_offset, count)
     VHDX_REGION_TABLE_ENTRY.decode(@vhdx_file.read(SIZEOF_VHDX_REGION_TABLE_ENTRY))
   end
 
+  def bat(entry)
+    @bat.empty? && process_bat
+    @bat[entry]
+  end
+
```

Comment:
@roliveri i had considered much of what you suggest.  I have no problem with it.  The question about `post_process_metadata` getting called before calling `bat` is a good one.  I have to walk through the logic again to be sure it is not possible.  Let's hold off on this until I have an answer.